### PR TITLE
Start angle for CanvasRenderingContext2D.createConicGradient() varies across browsers

### PR DESCRIPTION
**Pull request by Wish-Lin**
The description of start angle is incorrect. It starts at the right, not the top.

### CODE REVIEW

**Comment by wbamberg** [comment]
From the example it looks like the page is correct: https://pr22363.content.dev.mdn.mozit.cloud/en-US/docs/Web/API/CanvasRenderingContext2D/createConicGradient#filling_a_rectangle_with_a_conic_gradient .

The angle starts from a line going vertically up from the center (or we could say at 12 o'clock) and goes clockwise. Doesn't it?

**Comment by Wish-Lin** [comment]
> From the example it looks like the page is correct: https://pr22363.content.dev.mdn.mozit.cloud/en-US/docs/Web/API/CanvasRenderingContext2D/createConicGradient#filling_a_rectangle_with_a_conic_gradient .
> 
> The angle starts from a line going vertically up from the center (or we could say at 12 o'clock) and goes clockwise. Doesn't it?

Hmm? That's weird, since [this](https://imgur.com/a/LSdg3Mx) is what I see on all of my devices (Chrome & Samsung Internet)

**Comment by wbamberg** [comment]
> > From the example it looks like the page is correct: https://pr22363.content.dev.mdn.mozit.cloud/en-US/docs/Web/API/CanvasRenderingContext2D/createConicGradient#filling_a_rectangle_with_a_conic_gradient .
> > The angle starts from a line going vertically up from the center (or we could say at 12 o'clock) and goes clockwise. Doesn't it?
> 
> Hmm? That's weird, since [this](https://imgur.com/a/LSdg3Mx) is what I see on all of my devices (Chrome & Samsung Internet)

That is weird. I was looking at Firefox (and Safari is the same). On Chrome I see the same as you though...

**Comment by wbamberg** [comment]
https://bugs.chromium.org/p/chromium/issues/detail?id=1355539&q=conic-gradient&can=1 - it is reported as fixed, but I don;t know what version this would be in. We should update BCD for Chrome with this, but it would be good to know which version it is fixed in.

**Comment by wbamberg** [comment]
Actually it looks as if Chrome is following the spec here, and Firefox and Safari are not.

https://github.com/whatwg/html/pull/8213 seems to be a PR to update the spec to align with Firefox/Safari, but it was abandoned, as was the update to Chromium: https://chromium-review.googlesource.com/c/chromium/src/+/3846302 , as far as I can tell.

**Comment by Wish-Lin** [comment]
Well, it seems like a dilemma: Whether it should be aligned to its CSS counterpart or other Canvas methods. Before the problem is resolved, maybe we should just check the full scope of the problem and mention it on the page?

**Comment by Wish-Lin** [comment]
Hello?

**Comment by wbamberg** [comment]
Sorry @Wish-Lin . What we should do is:
- determine what the specced behavior is, and which browsers are not spec-compliant. I think the Chrome behavior is spec-compliant but would like confirmation
- update the page to document the specced behavior
- update the browser compatibility for the page to note the different behavior in the non-compliant browsers

@nt1m , I think you were involved in the speccing of this, can you confirm that the gradient is supposed to start at 3 o'clock 
(like the other Canvas methods, and as implemented by Chrome) and not at 12 o'clock (like the CSS version, and as implemented by Firefox and Safari)?

**Comment by nt1m** [comment]
Note that Safari just switched to match Chrome, see https://bugs.webkit.org/show_bug.cgi?id=244254 . Matching other canvas APIs does make sense to me.

**Comment by wbamberg** [comment]
Thanks @nt1m . Then I think we should document the Chrome behavior in the main page and note the other versions in BCD.

**Comment by carolhmj** [comment]
Would be nice to have consistent behavior between browsers as we are adding gradients to Babylon GUI https://github.com/BabylonJS/Babylon.js/pull/13361 and they look different on FF now 😅

**Comment by teoli2003** [comment]
> Would be nice to have consistent behavior between browsers as we are adding gradients to Babylon GUI [BabylonJS/Babylon.js#13361](https://github.com/BabylonJS/Babylon.js/pull/13361) and they look different on FF now 😅

Yes. but we are merely documenting the web platform here, not defining it. (You'll need to nag Mozilla in this case, if they are not yet aware)

**Comment by wbamberg** [comment]
I just filed https://github.com/mdn/browser-compat-data/pull/18734 to document Firefox/Safari behavior.

File: files/en-us/web/api/canvasrenderingcontext2d/createconicgradient/index.md [hunk 1]
```diff
@@ -30,7 +30,8 @@ createConicGradient(startAngle, x, y)
 ### Parameters
 
 - `startAngle`
-  - : The angle at which to begin the gradient, in radians. Angle measurements start vertically above the center and move around clockwise.
+  - : The angle at which to begin the gradient, in radians.
+  - Note: As of November 2022, the start angle of createConicGradient() varies across different browsers. Chromium-based browsers such as Chrome, Edge, Opera and Samsung Internet follow the `CanvasRenderingContext2D` tradition and start at 3 o'clock, while other browsers such as Firefox and Safari follow the CSS tradition and starts at 12 o'clock.
```
Comment (wbamberg) [inline files/en-us/web/api/canvasrenderingcontext2d/createconicgradient/index.md:34]:
```suggestion
  - : The angle at which to begin the gradient, in radians. The angle starts from a line going horizontally right from the center.
```

Comment (teoli2003) [reply files/en-us/web/api/canvasrenderingcontext2d/createconicgradient/index.md:34]: Yes, sounds good to me. I'm approving the PR here, but leave up to you to decide when to land it.

**Review by teoli2003 — APPROVED** [review]

**Comment by wbamberg** [comment]
Looks like Firefox is also spec-compliant now: https://bugzilla.mozilla.org/show_bug.cgi?id=1711171.

File: files/en-us/web/api/canvasrenderingcontext2d/createconicgradient/index.md [hunk 2]
```diff
@@ -22,7 +22,7 @@ createConicGradient(startAngle, x, y)
 ### Parameters
 
 - `startAngle`
-  - : The angle at which to begin the gradient, in radians. Angle measurements start vertically above the center and move around clockwise.
+  - : The angle at which to begin the gradient, in radians. The angle starts from a line going horizontally right from the center.
```
Comment (wbamberg) [inline files/en-us/web/api/canvasrenderingcontext2d/createconicgradient/index.md:25]:
```suggestion
  - : The angle at which to begin the gradient, in radians. The angle starts from a line going horizontally right from the center, and proceeds clockwise.
```

**Review by wbamberg — APPROVED** [review]
👍 I think we can merge this. Thank you for the PR, @Wish-Lin !

**Pull request merged**